pkgs/development/tools/fastddsgen/default.nix

```diff
@@ -33,12 +42,12 @@ stdenv.mkDerivation {
 
     gradle install --install_path=$out
 
-    # Override the default start script to use absolute java path
-    cat  <<EOF >$out/bin/fastddsgen
-    #!${runtimeShell}
-    exec ${openjdk17}/bin/java -jar "$out/share/fastddsgen/java/fastddsgen.jar" "\$@"
-    EOF
-    chmod a+x "$out/bin/fastddsgen"
+    # Override the default start script to use absolute java path.
+    # Make the unwrapped "cpp" available in the path, since the wrapped "cpp"
+    # passes additional flags and produces output incompatible with fastddsgen.
+    makeWrapper ${openjdk17}/bin/java $out/bin/fastddsgen \
+      --add-flags "-jar $out/share/fastddsgen/java/fastddsgen.jar" \
+      --set PATH ${lib.makeBinPath [ stdenv.cc.cc ]}
```

Comment:
I'm not sure hardcoding PATH to a single directory is a good thing. Are you sure that `fastddsgen` never tries to exec() anything else than `cpp`? I think it would be safer to use `--prefix` instead of `--set`.

Reply:
Changed to `--prefix` as suggested.